### ISSUE TITLE
restrict-plus-operands false positive for equal sign

### ISSUE THREAD

**Issue by buu700** [palantir/tslint#2443]
### Bug Report

This rule seems to incorrectly raise an error on every line of our code where an equal sign is used, and possibly more (it's too many lines to go through to see if other things may be triggering it).

- __TSLint version__: `5.0.0`
- __TypeScript version__: `2.2.2`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
// This triggers it.
const balls = true;

// This also triggers it.
return balls === true;
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"restrict-plus-operands": true
	}
}
```

#### Actual behavior

`Operands of '+' operation must either be both strings or both numbers`

#### Expected behavior

No error, as with tslint 4.x.
